### PR TITLE
Update pylint config to the latest version

### PR DESCRIPTION
**Pull request by victorvorobev**
Our config is outdated for a long time. Now the new default config us utilized with our updates ported into this config.

### CODE REVIEW

**Review by iamsaywhat — COMMENTED** [review]
Have you used `pylint --generate-rcfile` in 9e07b30c2cc3dfdc787a6d78a44c57136bdde364 to generate the default config? How about mentioning the command in the commit description? Just to make it clear exactly how this configuration appeared.

**Comment by victorvorobev** [comment]
@iamsaywhat I've actually used example from [pylint github repo](https://github.com/PyCQA/pylint/blob/main/examples/pylintrc) and manually applied our changes later. Didn't knew that I can create one myself, actually, need to give it a try

**Review by iamsaywhat — APPROVED** [review]
Just a few nitpick:
1) 9e07b30c2cc3dfdc787a6d78a44c57136bdde364 has redundant colon at the end of commit title
2) in 9e07b30c2cc3dfdc787a6d78a44c57136bdde364 you say that this is the default configuration, but if I understood you correctly from the comment above here you also applied some personal tweaks. Can we split it into two commits?
3) And It still will be nice to mention the way of getting default config (the link to the pylint repository or a command - it doesn't matter)

Otherwise, thank you! Finally, someone did it! 🙂

**Comment by victorvorobev** [comment]
@iamsaywhat 
Good points! I've updated the PR, please take another look

**Review by iamsaywhat — APPROVED** [review]
Great job!

File: python/pylintrc [hunk 1]
```diff
@@ -562,6 +316,316 @@ min-public-methods=0
...
+# Tells whether to store unknown words to the private dictionary (see the
+# --spelling-private-dict-file option) instead of raising a message.
+spelling-store-unknown-words=no
+
+
+[STRING]
+
+# This flag controls whether inconsistent-quotes generates a warning when the
+# character used as a quote delimiter is used inconsistently within a module.
+check-quote-consistency=no
```
Comment (dskorykh) [inline python/pylintrc:541]: Maybe it's worth enabling this one? Or it just will be checked in another place?

Comment (victorvorobev) [reply python/pylintrc:541]: Well, in our repos we have it covered with `pylint-quotes` plugin anyway, so shouldn't be a big deal

**Review by dskorykh — APPROVED** [review]

**Review by glebignatieff — APPROVED** [review]
Great job!

**Pull request merged**